pkg/api/steve/proxy/proxy.go

```diff
@@ -146,11 +147,24 @@ func (h *Handler) authLocalCluster(router *gmux.Router) func(rw http.ResponseWri
 func (h *Handler) matchManagementCRDs() gmux.MatcherFunc {
 	return func(req *http.Request, match *gmux.RouteMatch) bool {
 		splitPath := strings.Split(req.URL.Path, "/")
-		if len(splitPath) < 3 {
+		var resource string
+		// Extract either a cluster-wide or namespaced resource from the URL.
+		// The matcher function doesn't match paths like /api/v1/pods, which represent core Kubernetes resources.
+		// It also doesn't match those exempt from further authentication - only management resources and
+		// only those found in a special allow-list. The following structure is thus expected for these resources' paths.
+		switch len(splitPath) {
+		case 5: // Management resource at the cluster level, such as /apis/management.cattle.io/v3/podsecurityadmissionconfigurationtemplates
+			resource = splitPath[4]
+		case 7: // Management resource in a namespace, such as /apis/management.cattle.io/v3/namespaces/default/projectroletemplatebindings
```

Comment:
I think there's two other cases that need to be considered - requesting a specific resource (namespaced and non-namespaced). In these cases they will be one length longer than list path (e.x. `/apis/management.cattle.io/v3/namespaces/default/projectroletemplatebindings/my-prtb` or `/apis/management.cattle.io/v3/clusters/my-cluster`).

It might also be good to change the default case here. Right now, if we can't determine the path length, we return false (which defaults to allowing the request). This differs from the production implementation, which if it can confirm the use of a management.cattle.io api group (which it generally can since it's always at index 3) defaults to deny. If you go this route you'll also need to make sure that the schema discovery url (`/apis/management.cattle.io/v3`) works, but outside of that this might be overall safer.